include/shards.h

```diff
@@ -94,19 +94,22 @@ SH_ARRAY_DECL(SHPayloadArray, struct SHVarPayload);
 struct SHVar;
 SH_ARRAY_DECL(SHSeq, struct SHVar);
 
+struct SHTableImpl;
+struct SHSetImpl;
```

Comment:
What's the difference between doing this and `typedef void *SHTableImpl`.
Does C++ offers more guarantees in the former case?

For instance we also have `typedef void *SHPointer` that as far as I can tell is only used for `SHObject`. So should we have `struct SHObjectImpl` instead?

Reply:
Because I'm actually implementing these inside foundation.hpp they need to be forward declared as structs

Reply:
And the debugger will show the final definitions which are just std's maps and sets